lib/FrontendTool/FrontendTool.cpp

```diff
@@ -592,26 +594,33 @@ static bool performCompile(std::unique_ptr<CompilerInstance> &Instance,
   if (Invocation.getSILOptions().LinkMode == SILOptions::LinkAll)
     performSILLinking(SM.get(), true);
 
-  {
-    SharedTimer timer("SIL verification (pre-optimization)");
-    SM->verify();
+  if (!PrimarySourceFile) {
+    SM->linkAll();
+    SM->dump();
```

Comment:
@atrick That is what I said below. They should add a different flag for this. @jrose-apple This brings up an interesting question. Right now we do not have that concept in our pipeline. Optimizations that run at -Onone are considered mandatory.